.github/workflows/ci.yaml

```diff
@@ -16,19 +16,99 @@ on:
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-java@v2
+    - uses: actions/checkout@v4
+    - uses: actions/setup-java@v3
       with:
-        java-version: '11'
+        java-version: '17'
         distribution: 'temurin'
     - uses: skjolber/maven-cache-github-action@v1
       with:
         step: restore
     - run: mvn -B -U clean verify
+    - uses: actions/upload-artifact@v3
+      with:
+        name: cryostat-core
+        path: /home/runner/work/cryostat-core/cryostat-core/target/cryostat-core-2.24.0-SNAPSHOT.jar
     - uses: skjolber/maven-cache-github-action@v1
       with:
         step: save
+
+  cryostat-test:
+    needs: [build]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    uses: ./.github/workflows/cryostat-IT.yml
+    with:
+      build-arch: ${{ matrix.arch }}
+      skip-itests: ${{ matrix.arch != 'amd64' }}
+  
+  push-to-ghcr:
+    runs-on: ubuntu-latest
+    needs: [cryostat-test]
```

Comment:
How about building the image and push it to `ghcr.io` first? Then, pull it and use it the integration test?